### PR TITLE
Substitued XLEN with MXLEN

### LINKED ISSUE #52: Rename XLEN to MXLEN

### PR DESCRIPTION
**Pull request by AFOliveira**
Closes #52 . I've built the manual with the updates and nothing was broken.

### CODE REVIEW

File: backends/cfg_html_doc/templates/csr.adoc.erb [hunk 1]
```diff
@@ -100,7 +100,7 @@ This field has special behavior when written by software (_e.g._, through `csrrw
 +
 When software tries to write `csr_value`, the field will be written with the return value of the function below.
 +
-<%- if arch_def.multi_xlen? && csr.defined_in_all_bases? && field.defined_in_all_bases? -%>
+<%- if arch_def.multi_mxlen? && csr.defined_in_all_bases? && field.defined_in_all_bases? -%>
```
Comment (dhower-qc) [inline backends/cfg_html_doc/templates/csr.adoc.erb:103]: multi_xlen should stay multi_xlen; it is indicating whether or not, in some privilege mode, the xlen can change between 32 and 64 (not M-mode specific)

File: backends/cfg_html_doc/templates/inst.adoc.erb [hunk 2]
```diff
@@ -109,17 +109,17 @@ RV64::
...
 
-<%- xlens = inst.base.nil? ? (arch_def.multi_xlen? ? [32, 64] : [arch_def.mxlen]) : [inst.base] -%>
+<%- xlens = inst.base.nil? ? (arch_def.multi_mxlen? ? [32, 64] : [arch_def.mxlen]) : [inst.base] -%>
 
 <%- if inst.key?("operation()") -%>
 [tabs]
 ====
-<%- xlens.each do |effective_xlen| -%>
-Pruned, XLEN == <%= effective_xlen %>::
+<%- xlens.each do |effective_mxlen| -%>
```
Comment (dhower-qc) [inline backends/cfg_html_doc/templates/inst.adoc.erb:117]: stay effective_xlen

File: ext/riscv-opcodes
Comment (dhower-qc) [inline ext/riscv-opcodes:1]: Did you mean to check in this update to riscv-opcodes?

File: lib/arch_obj_models/csr.rb [hunk 3]
```diff
@@ -49,7 +49,7 @@ def defined_in_all_bases? = @data["base"].nil?
 
   # @param arch_def [ArchDef] A configuration
   # @return [Boolean] Whether or not the format of this CSR changes when the effective XLEN changes in some mode
-  def format_changes_with_xlen?(arch_def)
+  def format_changes_with_mxlen?(arch_def)
```
Comment (dhower-qc) [inline lib/arch_obj_models/csr.rb:52]: This should stay format_changes_with_xlen; it's not M-mode specific

**Comment by AFOliveira** [comment]
Thank you so much for the careful review! I'll try to address them all and get back to you in the next days in order to properly fix the PR.

File: ext/riscv-opcodes
Comment (AFOliveira) [reply ext/riscv-opcodes:1]: No. I believe it should be updated if we're using it but it should not have been commited here. I'll also remove it and add it when we start moving instructions from auto-instr.

**Comment by dhower-qc** [comment]
Closing for now; we are going to circle back to this once we have the "offical" list of parameters in hand.

**Pull request closed without merging**